jekyll/_cci2/google-container-engine.md

```diff
@@ -1,75 +1,110 @@
+### Authenticate to Google's Container Registry
 
-* `GOOGLE_PROJECT_ID`: the ID of your GCP project
-* `GOOGLE_CLUSTER_NAME`: the cluster to which deployments will occur
-* `GOOGLE_COMPUTE_ZONE`: which compute zone to use by default, e.g. `us-central1-a`
+If you chose the `google/cloud-sdk` image,
+no authentication is needed since this is a public image.
 
-### Setting Up a Job Step to Decode Credentials
-Once the `GOOGLE_AUTH` environment variable has been saved to your project, it will be readily available for use in the steps of your job's primary container. 
+```yaml
+version: 2
+jobs:
+  deploy-job:
+    docker:
+      - image: google/cloud-sdk
+```
 
-**config.yml**
+If you are using a custom image,
+authenticate to Google's Container Registry (GCR).
+Since you are [using a JSON key file](https://cloud.google.com/container-registry/docs/advanced-authentication#using_a_json_key_file),
+use the `auth` key in `config.yml`
+to specify credentials:
+
+```yaml
+version: 2
+jobs:
+  deploy-job:
+    docker:
+      - image: gcr.io/project/image-name
+        auth:
+          username: _json_key  # default username when using a JSON key file to authenticate
+          password: $GCLOUD_SERVICE_KEY  # encoded service account you created
+```
 
-```yml
-docker: 
-  - image: gcr.io/project/image-name
-    auth:
-    #Put the contents of keyfile.json into an environment variable for the build called GCR_CREDS, which is then passed in.
-      username: _json_key
-      password: $GOOGLE_AUTH 
-  - steps:
-     # ...
-     - run:
-       name: Dump Google Cloud Credentials to file
-       command: echo ${GOOGLE_AUTH} > ${HOME}/gcp-key.json
-     # ...  
+### Add a Job Step to Decode Credentials
+
+To authenticate the `gcloud` tool,
+add a job step to decode the `GCLOUD_SERVICE_KEY` environment variable:
+
+```yaml
+version: 2
+jobs:
+  deploy-job:
+    docker:
+      - image: google/cloud-sdk
+    steps:
+      - run:
+        name: Decode and Store Service Account
+        command: echo $GCLOUD_SERVICE_KEY | base64 --decode --ignore-garbage > ${HOME}/gcloud-service-key.json
 ```
 
-## Configuring `gcloud`
+### Configure `gcloud`
 
-As part of your deployment commands, configure `gcloud` to use the newly-configured service account and set up the appropriate defaults:
+Finally, as part of your deployment commands,
+update `gcloud`, authenticate, and set appropriate defaults for your project:
 
-```sh
-gcloud auth activate-service-account --key-file ${HOME}/gcp-key.json
+```bash
+gcloud --quiet components update
+gcloud auth activate-service-account --key-file=${HOME}/gcloud-service-key.json
 gcloud --quiet config set project ${GOOGLE_PROJECT_ID}
 gcloud --quiet config set compute/zone ${GOOGLE_COMPUTE_ZONE}
 gcloud --quiet container clusters get-credentials ${GOOGLE_CLUSTER_NAME}
 ```
 
-
+Refer to the [Google Cloud](https://circleci.com/docs/2.0/deployment-integrations/#google-cloud) deploy example for further steps.
```

Comment:
Feels weird to link people to the rest of deploy section, when the page itself is called "deploying to", but its what the deployment-integrations page is for so w/e